crates/astria-sequencer/src/app.rs

```diff
@@ -372,37 +380,50 @@ impl App {
                 .filter_map(Action::as_sequence)
                 .fold(0usize, |acc, seq| acc + seq.data.len());
 
-            // Don't include tx if it would make the sequenced block data too large.
-            if block_sequence_data_bytes + tx_sequence_data_bytes
-                > MAX_SEQUENCE_DATA_BYTES_PER_BLOCK
-            {
+            // don't include tx if it would make the sequenced block data too large
+            if !block_size_constraints.sequencer_has_space(tx_sequence_data_bytes) {
                 debug!(
                     transaction_hash = %telemetry::display::base64(&tx_hash),
-                    included_data_bytes = block_sequence_data_bytes,
+                    max_sequencer_bytes = block_size_constraints.max_sequencer,
+                    included_data_bytes = block_size_constraints.current_sequencer,
                     tx_data_bytes = tx_sequence_data_bytes,
-                    max_data_bytes = MAX_SEQUENCE_DATA_BYTES_PER_BLOCK,
                     "excluding transaction: max block sequenced data limit reached"
                 );
                 excluded_tx_count += 1;
                 continue;
             }
 
-            // store transaction execution result, indexed by tx hash
-            match self.deliver_tx(signed_tx.clone()).await {
-                Ok(events) => {
-                    self.execution_result.insert(tx_hash.into(), Ok(events));
+            // don't include tx if it would make the cometBFT block too large
+            if !block_size_constraints.cometbft_has_space(tx.len()) {
+                debug!(
+                    transaction_hash = %telemetry::display::base64(&tx_hash),
+                    max_cometbft_bytes = block_size_constraints.max_cometbft,
+                    included_data_bytes = block_size_constraints.current_cometbft,
+                    tx_data_bytes = tx.len(),
+                    "excluding transactions: max cometBFT data limit reached"
+                );
+                excluded_tx_count += 1;
+                continue;
+            }
```

Comment:
i would move this to directly under the `for` statement (also move `let tx_hash = Sha256::digest(&tx);` right above this block), since we know if the cometbft max size has been reached, the rest of the txs will be skipped also, so we don't need to spend resources decoding them

Reply:
I keep on going back and forth on if to short circuit the loop or not. If one transaction doesn't fit, others may still be able to. If we're trying to optimize for largest block (and max profit) it'd be better if we keep on checking. This isn't super relevant for this PR because the commitment size is only like 64 bytes and at most will kick out a single transaction. 

This will probably have to be refactored anyways once we start adding ToB bundles from the relay, where we'll be more likely to hit the max size.

Reply:
Put it up at the top!